<a href="https://colab.research.google.com/github/Nissun662/Data-Visual/blob/main/Quiz_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Visualization
Nishan Sunar
Assignment: 3 

Today's Assignment:
1. Start a notebook getting BigQuery to work. Feel free to use the authentication atop.
2. Navigate to the dataset 'austin_bikeshare.bikeshare_trips'
3. Compute how many entries are in the dataset
4. Compute the longest trip from 'duration_minutes'
5. Compute the average time for a trip


#Before we begin
We have to use Cloud Resource Manager to create a Cloud Platform project. After that we will enable billing and enable BigQuery APIs for the new froject created in Cloud Resourse Manager.

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


# Authentication Dataset
The above command does the authentication for my data set in my google cloud platform. 

# Using SQL 
In this assignment we use SQL because it is possible to illustrate huge data set in Database server. We can also use google cloud colab to do so.

I have created a data set in my BigQuery which is named : 
bustling-cosmos-338722.
I am going to run the command to compute the number of entries in the dataset.

As per the instructions we have to pull the dataset from Austin bikeshare trips and calculate the dataset compute the longest trip from 'duration_minutes' and compute the average time for the trip. 

#Computing the Number of Entries in the Dataset

The following command does the job. I have to write the name of the project and run the program to find out how many entries we have.

In [7]:
%%bigquery --project bustling-cosmos-338722 
SELECT 
  COUNT(*) as total_entries
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_entries
0        1424786

Therefore we have 1,424,786 entries in the dataset of Austin bikeshare 

# Computing the Longest Trip 


In [5]:
%%bigquery --project bustling-cosmos-338722 
 SELECT 
  MAX(duration_minutes)
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

f0_
0  34238

Hence, the longest trip will be 34,238 minutes.

#Computing average time for the trip


In [8]:
%%bigquery --project bustling-cosmos-338722
SELECT AVG(duration_minutes) 
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table

f0_
0  30.870428

Therefore, the average time for the trip will be approximately 31 minutes.